### PR TITLE
set up environment for Python virtualenv

### PR DESCRIPTION
**Pull request by kevinushey**
This PR adds a basic level of support for Python virtual environments.

The basic workflow (as I understand it) for Python virtual environments is to basically just call a script called `bin/activate`, which sets up some environment variables (notably, the `VIRTUAL_ENV` and `PATH` environment variables).

In this PR, we basically mock that process, by executing the venv'ed Python executable with those environment variables active.

### CODE REVIEW

File: src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp [hunk 1]
```diff
@@ -311,6 +314,50 @@ core::Error runChunk(const std::string& docId,
...
+      {
+         FilePath activatePath = enginePath.parent().childPath("activate");
+         if (activatePath.exists())
+         {
+            FilePath binPath = enginePath.parent();
+            FilePath venvPath = binPath.parent();
+            core::system::setenv(&env, "VIRTUAL_ENV", venvPath.absolutePath());
+
+#ifdef _WIN32
+            std::string kPathSeparator = ";";
```
Comment (jmcphers) [inline src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp]: Nit: It'd be nice if we could re-use the logic from `addToSystemPath`.

File: src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp [hunk 2]
```diff
@@ -311,6 +314,50 @@ core::Error runChunk(const std::string& docId,
    core::system::ProcessOptions options;
    options.terminateChildren = true;
    
+   core::system::Options env;
+   core::system::environment(env);
```
Comment (jmcphers) [inline src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp]: Shouldn't this be `core::system::environment(&env);`?

Comment (kevinushey) [inline src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp]: Yes, it should -- sorry, it looks like I missed that change!

**Comment by jjallaire** [comment]
One thing to note here is that once `reticulate` is on CRAN knitr / rmarkdown will probably depend on it which will allow us to do in-process execution of Python code and if we want to define a shared variable context. The only potential incompatibility this will cause is for documents which define a distinct engine for different chunks (since there can only ever be one Python interpreter inside an R process).

**Pull request merged**